Fix reactOnRailsPageUnloaded when there is no component on the page

NhanHo opened the pull request:
In `reactOnRailsPageLoaded`, if we finds no react component, the function [return ](https://github.com/NhanHo/react_on_rails/blob/f121906bf94a7f84ad4dde20c1cd7e93fa75f550/node_package/src/clientStartup.ts#L209) and `context.roots` is not initialized. This cause `findContext().roots` in `reactOnRailsPageUnloaded` to be undefined, and `for .. of` throws an exception.

This should mostly affect app using both hotwire and react_on_rails, as the recommended way to load js is to do the same file everywhere, otherwise the issue can be avoided by just not loading react_on_rails

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/shakacode/react_on_rails/1498)
<!-- Reviewable:end -->

davidalejandroaguilar commented:
Nice, I actually hit this while I was trying `react_on_rails` a couple days ago. It made me think it was unstable so decided not to use it.

justin808 commented on node_package/src/clientStartup.ts:
```diff
@@ -233,7 +233,12 @@ function unmount(el: Element): void {
+    // If no react on rails components
+    if (!roots) return;
+
+    for (const root of roots) {
```
Looks OK for me. @alexeyr-ci1 @alexeyr please review as you added the prior line for React 18.

This seems harmless, so I'll merge. Still, I'd like a post-merge review.

justin808 commented:
@NhanHo can you please add a CHANGELOG entry?

I'll merge as soon as this passes CI.

NhanHo commented:
@justin808 I fixed the linter issue and added changelog. If you decide to merge the other PR, you can close this one.

Thanks a lot for your work on Shakapacker and related projects.

justin808 commented:
@NhanHo Please merge https://github.com/shakacode/react_on_rails/pull/1499/files into your branch and I'll merge.

NhanHo commented:
@justin808 Updated

justin808 commented:
Thanks @NhanHo!
